advice_animal/runner.py

```diff
@@ -11,10 +15,162 @@
 LOG = logging.getLogger(__name__)
 
 
+def run_cmd(cmd: list[Union[str, Path]], check: bool = True) -> Tuple[str, int]:
+    LOG.info("Run %s in %s", cmd, os.getcwd())
+    proc = subprocess.run(cmd, encoding="utf-8", capture_output=True, check=check)
+    LOG.debug("Ran %s -> %s", cmd, proc.returncode)
+    LOG.debug("Stdout: %s", proc.stdout)
+    return proc.stdout, proc.returncode
+
+
+@dataclass
+class Result:
+    advice_name: str
+    success: bool
+    project: str = ""
+    message: str = ""
+    changes_needed: bool = False
+
+
 class Runner:
-    def __init__(self, env: Env, check_dir: Path) -> None:
-        self.env = env
-        self.check_dir = check_dir
+    def __init__(
+        self, advice_path: Path, inplace: bool, mode: Literal["check", "diff", "apply"]
+    ) -> None:
+        if not advice_path.is_dir():
+            raise ValueError(f"{advice_path} is not a directory")
+
+        if inplace and mode != "apply":
+            raise ValueError("inplace only valid with mode 'apply'")
+
+        self.advice_path = advice_path
+        self.inplace = inplace
+        self.mode = mode
+
+    def run(
+        self,
+        repo: Path,
+        confidence_filter: FixConfidence,
```

Comment:
I think you mentioned the idea of combining these three into some kind of "filter spec" object -- I like that idea.  It could even happen once where the ctx.obj is set up.

Reply:
Didn't want to add too many features in this PR.